vscode/src/commands/GhostHintDecorator.ts

```diff
@@ -94,21 +96,26 @@ function getSymbolDecorationPadding(
 type GhostVariant = 'EditOrChat' | 'Document' | 'Generate'
 type EnabledFeatures = Record<GhostVariant, boolean>
 
+/**
+ * NOTE: Ghost Text should be disabled for users in the HoverCommands A/B test treatment group.
+ */
 export async function getGhostHintEnablement(): Promise<EnabledFeatures> {
+    const [ghostTextFeatureFlag, hoverFeatureFlag] = await Promise.all([
+        featureFlagProvider.evaluateFeatureFlag(FeatureFlag.CodyDocumentHints),
+        featureFlagProvider.evaluateFeatureFlag(FeatureFlag.CodyHoverCommands),
+    ])
+
     const config = vscode.workspace.getConfiguration('cody')
     const configSettings = config.inspect<boolean>('commandHints.enabled')
     const settingValue = configSettings?.workspaceValue ?? configSettings?.globalValue
 
     // Return the actual configuration setting, if set. Otherwise return the default value from the feature flag.
     return {
         /**
-         * We're not running an A/B test on the "Opt+K to Text".
-         * We can safely set the default of this to `true`.
+         * Toggle the default settings for EditOrChat & Document based on the feature flagss for hover commands and ghost text.
          */
-        EditOrChat: settingValue ?? true,
-        Document:
-            settingValue ??
-            (await featureFlagProvider.evaluateFeatureFlag(FeatureFlag.CodyDocumentHints)),
+        EditOrChat: settingValue ?? (ghostTextFeatureFlag && !hoverFeatureFlag),
+        Document: settingValue ?? (ghostTextFeatureFlag && !hoverFeatureFlag),
```

Comment:
```suggestion
        EditOrChat: settingValue ?? !hoverFeatureFlag,
        Document: settingValue ?? !hoverFeatureFlag,
```

I think is what we want. That way 50% of users will get the hover experiment and the other 50% will get the ghost text